nexus/pkg/server/filepusher.go

```diff
@@ -113,12 +135,21 @@ func (u *Uploader) upload(task *UploadTask) error {
 		task.url,
 		file,
 	)
+
+	for _, header := range task.headers {
+		parts := strings.Split(header, ":")
```

Comment:
nit: should we already have the headers constructed with it's correct parts, so we don't need to do this computation?

Reply:
for sure.. will address more in the future.  list of tuple strings probably